src/v/kafka/server/handlers/fetch.cc

```diff
@@ -975,12 +976,16 @@ class nonpolling_fetch_plan_executor final : public fetch_plan_executor::impl {
     ss::future<> execute_plan(op_context& octx, fetch_plan plan) final {
         start_worker_aborts(plan);
 
-        co_await log_exceptions(do_execute_plan(octx, std::move(plan)));
+        co_await handle_exceptions(do_execute_plan(octx, std::move(plan)));
 
         // Send abort signal to workers and wait for all workers to end before
         // returning.
         co_await abort_workers().finally(
           [this] { return _workers_gate.close(); });
+
+        if (_thrown_exception) {
+            std::rethrow_exception(_thrown_exception);
```

Comment:
should this eat the exception pointer like `std::rethrow_exception(std::exchange(_thrown_exception, {}));` ?

Reply:
We could, not sure what we'd gain by doing so.  And at this point the exception thrown isn't caught by the fetch executor so during the stack unwinding the fetch executor and hence `_thrown_exception` will be freed so it shouldn't matter. Beyond that `std::exception_ptr` can roughly be treated as a shared pointer according to the standard and should remain valid even if rethrown.

Reply:
>executor so during the stack unwinding the fetch executor and hence _thrown_exception will be freed so it shouldn't matter.

Ah, right. Thanks!